V2/jong

Jongminfire opened the pull request:
## 개발 사항

1. 목업 데이터 대신 Redux 사물함 데이터로 컴포넌트 연결했습니다.
2. 사물함 버튼을 누르거나 Submit  버튼을 눌렀을 때 함수를 만들어서 사용자의 adminType이나 사물함의 status에 따라 조건을 두어 다른 이벤트를 처리하도록 했습니다.
3. 사물함을 신청할 때 Redux 데이터 바꾸도록 했습니다.

-> 사물함에 관련된 기능들 함수나 구조 변경이 필요할 수 있을 것 같은데 한번 확인 부탁드려요~

## 앞으로 해야 할 일

1. 사물함에 관련된 기능들에 Swal 추가

Jongminfire committed:
Redux 사물함 데이터 연결

Jongminfire committed:
사물함 버튼 기능추가

1. 사물함 버튼 신청시 Redux 데이터 변경
2. Redux 데이터 토대로 사물함 버튼 누를 때 조건 나눠서 다른 이벤트처리

Kyun2da commented on src/Components/Cabinet/Cabinet.tsx:
```diff
@@ -6,13 +6,9 @@ import { database } from '../../config/firebase.config';
 import CabinetButtons from '../CabinetButtons';
 
-export type CabinetProps = { userData: any };
+export type CabinetProps = { cabinetData: any };
```
가급적이면 any 타입은 지양해주시는게 좋을 것 같습니다. 타입스크립트를 쓰는 의미가 퇴색되어버리는 것 같아요.

Kyun2da commented on src/pages/MainPage/MainPage.tsx:
```diff
@@ -66,7 +32,7 @@ function MainPage({}: MainPageProps) {
         <MenuInfo />
       </Header>
-      <Cabinet userData={userData} />
+      <Cabinet cabinetData={cabinet} />
```
셀렉터를 받은거를 MainPage에서 props로 뿌려주지말고 CabinetComponent에서 셀렉터를 받으면 어떨까요?

Kyun2da commented on src/Components/CabinetButtons/CabinetButtons.tsx:
```diff
@@ -26,12 +33,92 @@ export type CabinetData = {
+      }
+    } else {
+      if (item[select].status === 0) {
+        database.ref(`cabinet/${index}/item/${select}`).set({
```
108번째 줄부터 117번째줄 함수 3개 firebase폴더로 빼서 firebase 데이터 등록 함수로 관리하는 것도 좋을 것 같습니다!

Kyun2da commented:
리뷰 확인하셨으면 머지하셔도 되고 고치고싶으신부분 있으시면 고치시고 머지하셔도 될것같습니다!

Jongminfire committed:
리뷰 내용 수정

1. any타입 사용 -> 타입 정의
2. Cabinet 컴포넌트에서 cabinet 데이터 받아옴
3. cabinet submit 함수 util/firebase 폴더로 분리
